testing/e2e/skr-tester/pkg/command/assert.go

```diff
@@ -135,6 +146,110 @@ func (cmd *AssertCommand) Run() error {
 			}
 		}
 		fmt.Println("All specified admins are found in cluster role bindings")
+	} else if cmd.btpManagerSecretExists { //
+		kubeconfig, err := kcpClient.GetKubeconfig(cmd.instanceID)
+		if err != nil {
+			return fmt.Errorf("failed to get kubeconfig: %v", err)
+		}
+		cmd.checkBTPManagerSecret(kubeconfig)
+	} else if cmd.deleteBtpManagerSecret {
+		kubeconfig, err := kcpClient.GetKubeconfig(cmd.instanceID)
+		if err != nil {
+			return fmt.Errorf("failed to get kubeconfig: %v", err)
+		}
+		restCfg, err := clientcmd.RESTConfigFromKubeConfig(kubeconfig)
+		if err != nil {
+			return fmt.Errorf("while creating rest config from kubeconfig: %w", err)
+		}
+		k8sCli, err := client.New(restCfg, client.Options{
+			Scheme: scheme.Scheme,
+		})
+		if err != nil {
+			return fmt.Errorf("while creating k8s client: %w", err)
+		}
+		secrets := &v1.SecretList{}
+		listOpts := []client.ListOption{
+			client.InNamespace("kyma-system"),
+			client.MatchingFields{"metadata.name": "sap-btp-manager"},
+		}
+		err = k8sCli.List(context.Background(), secrets, listOpts...)
+		if err != nil {
+			return fmt.Errorf("while getting secret from instance: %w", err)
+		}
+
+		if len(secrets.Items) != 1 {
+			return fmt.Errorf("found %d secrets but expected 1", len(secrets.Items))
+		}
+		secret := secrets.Items[0]
+		err = k8sCli.Delete(context.Background(), &secret)
+		if err != nil {
+			return fmt.Errorf("while deleting secret from instace: %w", err)
+		}
+		fmt.Println("BTP manager secret deleted successfully")
+		for i := 0; i < 10; i++ {
```

Comment:
Magic value - why 10?

Reply:
Changed to a less magic value. 10 retries because it is enough.